DeepSee_Stars/DeepSee_Stars/SceneManager/Scene/GameScene/Enemies/Enemy/BaseEnemyAction/BaseEnemyAction.h

```diff
@@ -0,0 +1,134 @@
+﻿#ifndef BASE_ENEMY_ACTION_H_
+#define BASE_ENEMY_ACTION_H_
+
+#include <deque>
+
+#include <d3dx9.h>
+
+#include "../../../Enum/Direction.h"
+
+namespace deepseestars
+{
+	enum ActionPattern
+	{
+		PATROL,
+		L_SHAPED,
+		STRAIGHTLINE,
+		DONTMOVE
+	};
+
+	/// <summary>
+	/// 敵の初期位置や移動範囲などの最初に必要なデータをまとめた構造体
+	/// </summary>
+	struct EnemyMovingData
+	{
+	public:
+		D3DXVECTOR2		m_initPos;			//初期位置
+		D3DXVECTOR2		m_movingPos;		//移動できる最大範囲
+		ActionPattern	m_actionPattern;	//行動パターン
+		bool			m_isClockWise;		//時計回りで巡回するか
+	};
+
+	/// <summary>
+	/// 敵の巡回パターンの基底クラス
+	/// </summary>
+	class BaseEnemyAction
+	{
+	public:
+		/// <summary>
+		/// 敵の移動に関する情報をまとめた構造体
+		/// </summary>
+		struct TranslationData
+		{
+		public:
+			D3DXVECTOR2   m_pos;
+			D3DXVECTOR2   m_speed = { 10.f,10.f };
+			D3DXVECTOR2   m_movement;
+			Direction     m_direction;			
+		};
+
+		/// <summary>
+		/// 巡回するためのデータを詰めるコンストラクタ
+		/// </summary>
+		/// <param name="translationData">敵クラスにある移動に関するデータの参照</param>
+		/// <param name="initPos">動き始める視点座標</param>
+		/// <param name="dest">目的地の絶対座標</param>
+		BaseEnemyAction(TranslationData& translationData,
+			const D3DXVECTOR2& initPos, const D3DXVECTOR2& dest) :
+			m_translationData(translationData),
+			m_initPos(initPos), m_movingRange(dest)
+		{}
+		
+		~BaseEnemyAction() {}
+
+		/// <summary>
+		/// 敵の更新処理
+		/// </summary>
+		virtual void Update() = 0;
+
+	protected:
+		/// <summary>
+		/// 座標の更新をする
+		/// </summary>
+		void UpdatePos()
+		{
+			m_translationData.m_speed = m_distance / m_MOVE_SPEED_FRAME;
+			m_translationData.m_movement = m_translationData.m_speed;
+
+			m_translationData.m_pos += m_translationData.m_movement;
+		}
+
+		/// <summary>
+		/// 目的地に着いたときの処理
+		/// </summary>
+		void ArriveAtDestination(D3DXVECTOR2* pDest)
+		{
+			int currentDest = m_destQueue.front();
+			D3DXVECTOR2 absSpeed = { fabsf(m_translationData.m_speed.x),fabsf(m_translationData.m_speed.y) };
+
+			D3DXVECTOR2 topLeft = { pDest[currentDest].x - absSpeed.x,pDest[currentDest].y - absSpeed.y };
+			D3DXVECTOR2 bottomRight = { pDest[currentDest].x + absSpeed.x,pDest[currentDest].y + absSpeed.y };
+
+			//目的地の範囲内に入っていたら
+			if (((topLeft.x <= m_translationData.m_pos.x) && (bottomRight.x >= m_translationData.m_pos.x)) &&
```

Comment:
```suggestion
			if (((topLeft.x <= m_translationData.m_pos.x) && (m_translationData.m_pos.x <= bottomRight.x )) &&
```
好みの問題ですがこちらのほうがグラフとして認識しやすいと思います